tests/interactive/test_app.py

```diff
@@ -81,17 +82,65 @@ async def test_input_and_buttons():
         }
         """
         )
+
+        # press clear input button
+        await pilot.click("#clear_input_button")
+
+        # assert that the curent_module and test_module's are structurally equivalent
         await pilot.pause()
-        assert (
-            app.input_text_area.text
-            == """
+        assert app.input_text_area.text == ""
+
+        # Testing a pass
+        app.input_text_area.insert(
+            """
         func.func @hello(%n : index) -> index {
           %two = arith.constant 2 : index
           %res = arith.muli %n, %two : index
           func.return %res : index
         }
         """
         )
-        await pilot.click("#clear_input_button")
+
+        # Select a pass
+        app.passes_selection_list.select(
+            convert_func_to_riscv_func.ConvertFuncToRiscvFuncPass
+        )
+
         await pilot.pause()
-        assert app.input_text_area.text == ""
+        assert (
+            str(app.current_module)
```

Comment:
this should be using structural equality

Reply:
There's an example of this just above BTW, there's a bit more code to write for this module but it should be pretty similar. I'd probably recommend changing the example being tested to something like `dce` on arith. The riscv lowering representation is likely to change, and it's a nice general principle to make unit tests be as specific as possible to avoid breaking tests that are testing something unrelated